pkg/controller/metrics/metrics.go

```diff
@@ -339,6 +395,65 @@ func (mc *Calculator) Start(ctx context.Context) error {
 	return nil
 }
 
+// registerOptionalMetrics registers the metrics, and updates their configs in the corresponding maps
+func registerOptionalMetrics(mConfig *metricsconfig.MetricsConfig) {
+	if mConfig.MetricsWithDuration != nil {
+		if MapMetricToDuration == nil {
+			MapMetricToDuration = make(map[*prometheus.HistogramVec]time.Duration)
+		}
+		for _, metric := range mConfig.MetricsWithDuration {
+			switch metric.Name {
+			case metricsconfig.CurrentStopping:
+				// Metrics can only be registered once. Unregistering and registering again doesn't work well because
+				// unregistering is only meant for rare unit tests and doesn't clear all the records of that metric
+				if !existsInDurationMetricsMap(MetricStoppingClustersSeconds) {
```

Comment:
So - my pod kept on Crashing because without the check here, it was executing `MustRegister` every reconcile, which threw the `already registered` error.
There's a `metrics.Registry.Unregister` which I tried using, but that doesn't clear all the records, hence the `already registered` error wasn't cleared. So I've put in this hack to ensure we only register once.
Whenever the hiveconfig is edited, it anyways terminates the pod and starts a new instance, so I didn't need to clear any values - this hack is only necessary for the reconciles.
I've broken this entire registering part into a new function for ease of reading code.

Reply:
Have also tested with updating the existing duration in hiveconfig, as well as deleting/adding metrics - terminates pod everytime and creates a new instance - but you can still see the values of the older instance on the prometheus, so we should be good